mp/src/game/client/momentum/ui/HUD/hud_speedometer.cpp

```diff
@@ -423,148 +321,138 @@ void CHudSpeedMeter::OnThink()
             {
-                // Really gross way of ripping apart this string and
-                // making some sort of quasi-frankenstein string of the float as a rounded int
-                char firstThree[3];
-                Q_strncpy(firstThree, enterVelComparisonUnrounded, 3);
-                const char *compFloat = enterVelComparisonUnrounded;
-                Q_snprintf(enterVelComparisonRounded, BUFSIZELOCL, " %s %i)", firstThree,
-                           RoundFloatToInt(Q_atof(compFloat + 3)));
-
-                // Update the compare color to have the alpha defined by animations (fade out if 5+ sec)
-                compareColorFade.SetColor(compareColor.r(), compareColor.g(), compareColor.b(), m_fStageStartAlpha);
-
+                m_pStageEnterExitComparisonLabel->SetPos(0, m_pStageEnterExitLabel->GetYPos());
+                m_pStageEnterExitComparisonLabel->SetFgColor(Color(compareColor.r(), compareColor.g(), compareColor.b(), m_fStageStartAlpha));
+                m_pStageEnterExitComparisonLabel->SetText(enterVelComparisonStr);
                 // Add the compare string to the length
-                combinedLength += spaceBetweenLabels + UTIL_ComputeStringWidth(labelFont, enterVelComparisonRounded);
+                combinedLength += UTIL_ComputeStringWidth(labelFont, enterVelComparisonStr);
+            }
+            else
+            {
+                // ignore ending space as comparison is not being shown
+                combinedLength -= UTIL_ComputeStringWidth(labelFont, " ");
             }
-            int offsetXPos = 0 - ((GetWide() - combinedLength) / 2);
 
-            // split_xpos = GetWide() / 2 - (enterVelANSIWidth + increaseX) / 2;
+            int offsetXPos = 0 - ((GetWide() - combinedLength) / 2);
             m_pStageEnterExitLabel->SetPos(offsetXPos, m_pStageEnterExitLabel->GetYPos());
-            m_pStageEnterExitLabel->SetFgColor(actualColorFade);
-            m_pStageEnterExitLabel->SetText(enterVelRounded);
+            m_pStageEnterExitLabel->SetFgColor(Color(enterColor.r(), enterColor.g(), enterColor.b(), m_fStageStartAlpha));
+            m_pStageEnterExitLabel->SetText(enterVelStr);
+        }
+    }
+}
 
-            // print comparison as well
-            if (loadedComparison)
-            {
-                m_pStageEnterExitComparisonLabel->SetPos(spaceBetweenLabels, m_pStageEnterExitLabel->GetYPos());
-                m_pStageEnterExitComparisonLabel->SetFgColor(compareColorFade);
-                m_pStageEnterExitComparisonLabel->SetText(enterVelComparisonRounded);
-            }
-            else // only print velocity
-            {
-                m_pStageEnterExitComparisonLabel->SetText("");
-            }
+void CHudSpeedMeter::SetLabelHeight(bool isActive, Label *label, int height)
+{
+    label->SetAutoTall(isActive);
+    label->SetTall(isActive ? height : 0);
+    if (!isActive) label->SetText("");
```

Comment:
Do you need to clear it when it's inactive?

Reply:
If you mean the `if (!isActive) label->SetText("");` part yes. Had to add that as the text of the last known value was not being cleared when labels were inactivated. Was hoping setting it to 0 height would keep it from displaying.